File: python/easy/prob_002_valid_sequence/solution.py
def valid_sequence(array, sequence):
  sequence_index = 0
  for number in array:
    if sequence_index == len(sequence):
      return True
    if number == sequence[sequence_index]:
      sequence_index = sequence_index + 1
  return sequence_index == len(sequence)

print(valid_sequence([5, 1, 22, 25, 6, -1, 8, 10, 9], [1, 6, -1, 10]))
print(valid_sequence([5, 1, 22, 25, 6, -1, 8, 10, 9], [5, 1, 22, 777, 10]))

File: ruby/easy/prob_002_valid_sequence/solution.rb
def valid_sequence(array, sequence)
  sequence_index = 0

  array.each do |number|
    if sequence_index == sequence.length
      return true
    end

    if sequence[sequence_index] == number
      sequence_index = sequence_index + 1
    end
  end

  sequence_index == sequence.length
end

puts valid_sequence([5, 1, 22, 25, 6, -1, 8, 10, 9], [1, 6, -1, 10])
puts valid_sequence([5, 1, 22, 25, 6, -1, 8, 10, 9], [5, 1, 22, 777, 10])